Homepage panic when using spanish as preferred language

To reproduce: Start the app on a new dir, select Spanish on the startup settings page, then attempt to visit the wallet page.

After investigating the issue: I noticed we need to avoid the pattern of assigning translation strings to global variables. This is because the default language will be used when translating the strings the first time, and then it won't match subsequent translation output after the user has changed their language. e.g `navigationTabTitles` in home_page.go.

Also, changing a language causes panic if you attempt to visit the previous page (still in the pagestack but with outdated translation). 

To fix: I suggest we init or re-init dropdowns, nav items or segment controls (wherever a translated string is used in a struct type)  that were originally initiated in page constructors in `OnNavigatedTo` instead. That way we are sure the current translation string is used, thereby avoiding a panic when there's a mismatch in `switch` statements that decide which to page show using the output from `values.String(...)`.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x100f44e90]

goroutine 30 [running]:
github.com/crypto-power/cryptopower/app.(*PageStack).PushAndNavigate(0xc002fe14d0, {0x0, 0x0}, {0x101ddd730, 0xc002f68150})
        /Users/philemon/Desktop/osp/cryptopower/app/pagestack.go:46 +0x130
github.com/crypto-power/cryptopower/app.(*MasterPage).Display(0xc002f68150, {0x0?, 0x0?})
        /Users/philemon/Desktop/osp/cryptopower/app/masterpage.go:46 +0x33
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).displaySelectedPage(0xc003a1a680, {0xc0000cd8b5, 0x7})
        /Users/philemon/Desktop/osp/cryptopower/ui/page/root/home_page.go:290 +0x134
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0xc003a1a680)
        /Users/philemon/Desktop/osp/cryptopower/ui/page/root/home_page.go:191 +0x9f
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc003626960, {{0xc1603d777880a920, 0x193a26d0c, 0x102be7780}, {0x40000000, 0x40000000}, {0x640, 0x514}, {0x0, 0x0, ...}, ...})
        /Users/philemon/Desktop/osp/cryptopower/ui/window.go:229 +0x194
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc003626960)
        /Users/philemon/Desktop/osp/cryptopower/ui/window.go:200 +0x2b6
main.main.func3()
        /Users/philemon/Desktop/osp/cryptopower/main.go:103 +0x17
created by main.main in goroutine 1
        /Users/philemon/Desktop/osp/cryptopower/main.go:101 +0x448

```

EDIT: #387 will resolve this issue but re-translating strings created in page constructors would be a separate issue and PR.

--- a/ui/page/root/home_page.go
+++ b/ui/page/root/home_page.go
@@ -62,14 +62,6 @@ type HomePage struct {
 	totalBalanceUSD string
 }
 
-var navigationTabTitles = []string{
-	values.String(values.StrOverview),
-	values.String(values.StrTransactions),
-	values.String(values.StrWallets),
-	values.String(values.StrTrade),
-	values.String(values.StrGovernance),
-}
-
 func NewHomePage(dexCtx context.Context, l *load.Load) *HomePage {
 	hp := &HomePage{
 		Load:        l,
@@ -81,26 +73,8 @@ func NewHomePage(dexCtx context.Context, l *load.Load) *HomePage {
 	hp.hideBalanceButton = hp.Theme.NewClickable(false)
 	hp.appLevelSettingsButton = hp.Theme.NewClickable(false)
 	hp.appNotificationButton = hp.Theme.NewClickable(false)
-
-	hp.navigationTab = l.Theme.Tab(layout.Horizontal, false, navigationTabTitles)
-
 	_, hp.infoButton = components.SubpageHeaderButtons(l)
 	hp.infoButton.Size = values.MarginPadding15
-
-	hp.sendReceiveNavItems = []components.NavBarItem{
-		{
-			Clickable: hp.Theme.NewClickable(true),
-			Image:     hp.Theme.Icons.SendIcon,
-			Title:     values.String(values.StrSend),
-			PageID:    send.SendPageID,
-		},
-		{
-			Clickable: hp.Theme.NewClickable(true),
-			Image:     hp.Theme.Icons.ReceiveIcon,
-			Title:     values.String(values.StrReceive),
-			PageID:    ReceivePageID,
-		},
-	}
 
 	go func() {
 		hp.isConnected.Store(libutils.IsOnline())
@@ -124,10 +98,38 @@ func NewHomePage(dexCtx context.Context, l *load.Load) *HomePage {
 	}
 	hp.walletSelectorPage.showNavigationFunc = hp.showNavigationFunc
 
+	return hp
+}
+
+// initPageItems initializes navbar items that require the latest translation
+// string and MUST be called from OnNavigatedTo.
+func (hp *HomePage) initPageItems() {
+	navigationTabTitles := []string{
+		values.String(values.StrOverview),
+		values.String(values.StrTransactions),
+		values.String(values.StrWallets),
+		values.String(values.StrTrade),
+		values.String(values.StrGovernance),
+	}
+	hp.navigationTab = hp.Theme.Tab(layout.Horizontal, false, navigationTabTitles)
+
+	hp.sendReceiveNavItems = []components.NavBarItem{
+		{
+			Clickable: hp.Theme.NewClickable(true),
+			Image:     hp.Theme.Icons.SendIcon,
+			Title:     values.String(values.StrSend),
+			PageID:    send.SendPageID,
+		},
+		{
+			Clickable: hp.Theme.NewClickable(true),
+			Image:     hp.Theme.Icons.ReceiveIcon,
+			Title:     values.String(values.StrReceive),
+			PageID:    ReceivePageID,
+		},
+	}
+
 	hp.initBottomNavItems()
 	hp.bottomNavigationBar.OnViewCreated()
-
-	return hp
 }
 
 // ID is a unique string that identifies the page and may be used
@@ -143,6 +145,8 @@ func (hp *HomePage) ID() string {
 // Part of the load.Page interface.
 func (hp *HomePage) OnNavigatedTo() {
 	hp.ctx, hp.ctxCancel = context.WithCancel(context.TODO())
+
+	hp.initPageItems()
 
 	go hp.CalculateAssetsUSDBalance()
 	if !hp.AssetsManager.DexcReady() {
@@ -249,10 +253,10 @@ func (hp *HomePage) HandleUserInteractions() {
 	hp.floatingActionButton.CurrentPage = hp.CurrentPageID()
 	for _, item := range hp.bottomNavigationBar.BottomNavigationItems {
 		for item.Clickable.Clicked() {
-			hp.displaySelectedPage(item.Title)
 			if hp.ID() == hp.CurrentPageID() {
 				continue
 			}
+			hp.displaySelectedPage(item.Title)
 		}
 	}
 

